### PR TITLE
Internal Tracker

### PR DESCRIPTION
**Pull request by ubahnverleih**
* renamed `current_position` to `public_geolocation` in Bikes
* renamed `current_position` to `current_geolocation` in Locations
* filter for non internal trackers in `public_geolocation`
* added `internal_geolocation` in Bikes (does not filter out internal marked locations)
* show `public_geolocation` and `internal_geolocation` for Bikes in admin

### CODE REVIEW

File: bikesharing/migrations/0021_location_internal_location.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('bikesharing', '0020_locationtracker_internal_only'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='location',
+            name='internal_location',
```
Comment (robbi5) [inline bikesharing/migrations/0021_location_internal_location.py]: this is a field on the `location` model. we should not duplicate the model name in the field again, so lets use just `internal`? :)

File: bikesharing/migrations/0020_locationtracker_internal_only.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('bikesharing', '0019_auto_20200317_2042'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='locationtracker',
+            name='internal_only',
+            field=models.BooleanField(default=False, help_text="Internal trackers don't publish their locations to the enduser. They are usefull for backup trackers with lower accuracy e.g. wifi trackers."),
```
Comment (robbi5) [inline bikesharing/migrations/0020_locationtracker_internal_only.py]: typo: `useful`

File: bikesharing/models.py [hunk 3]
```diff
@@ -78,11 +78,19 @@ def __repr__(self):
...
-    def current_position(self):
+    def public_geolocation(self):
+        if not self.id:
+            return None
+        try:
+            return Location.objects.filter(bike=self, internal_location=False).latest('reported_at')
+        except ObjectDoesNotExist:
+            return None
+
+    def internal_geolocation(self):
```
Comment (robbi5) [inline bikesharing/models.py]: internal geolocation also returns locations that are not internal. is this on purpose?

File: bikesharing/migrations/0020_locationtracker_internal_only.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('bikesharing', '0019_auto_20200317_2042'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='locationtracker',
+            name='internal_only',
```
Comment (robbi5) [inline bikesharing/migrations/0020_locationtracker_internal_only.py]: personal naming opinion: I would drop the `_only` there.

File: bikesharing/admin.py [hunk 5]
```diff
@@ -31,21 +31,37 @@ class BikeAdmin(LeafletGeoAdmin, admin.ModelAdmin):
 
     @mark_safe
     def location(self, obj):
-        if obj is None or obj.current_position() is None:
```
Comment (robbi5) [inline bikesharing/admin.py]: this looks (maybe only in the diff) like a lot of copied code there. not sure if we should put this already in an function and call it twice with `obj.public_geolocation()` and `obj.internal_geolocation()`

File: bikesharing/models.py (hunk 3 above)
Comment (ubahnverleih) [reply bikesharing/models.py]: idea was to have definitiv the latest location. but open for discussion. In Bike i only display internal location, if it differs from the public location

File: bikesharing/migrations/0020_locationtracker_internal_only.py (hunk 2 above)
Comment (ubahnverleih) [reply bikesharing/migrations/0020_locationtracker_internal_only.py]: fixed in 4d82c86

File: bikesharing/migrations/0020_locationtracker_internal_only.py (hunk 4 above)
Comment (ubahnverleih) [reply bikesharing/migrations/0020_locationtracker_internal_only.py]: fixed in 9d62f3f

File: bikesharing/migrations/0021_location_internal_location.py (hunk 1 above)
Comment (ubahnverleih) [reply bikesharing/migrations/0021_location_internal_location.py]: renamed in 31b1b9d

File: bikesharing/admin.py (hunk 5 above)
Comment (ubahnverleih) [reply bikesharing/admin.py]: the truth is... that it is copied

File: bikesharing/models.py (hunk 3 above)
Comment (ubahnverleih) [reply bikesharing/models.py]: changed it now. I think its now less confusing for users

File: bikesharing/admin.py (hunk 5 above)
Comment (ubahnverleih) [reply bikesharing/admin.py]: hope thats better: 76bc7e673d0fd878a59f41a3cc3f96a39a728c50

**Pull request merged**